### PR TITLE
Adding model metadata tests

### PR DESCRIPTION
**Pull request by rasapala**

### CODE REVIEW

File: src/test/get_mediapipe_graph_metadata_response_test.cpp [hunk 1]
```diff
@@ -107,3 +122,207 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
...
+    class MockMediapipeGraphDefinitionGetInputsOutputsInfo : public ovms::MediapipeGraphDefinition {
+        ovms::Status status = ovms::StatusCode::OK;
+
+    public:
+        MockMediapipeGraphDefinitionGetInputsOutputsInfo() :
+            MediapipeGraphDefinition("mediaDummy", {}, {}) {
+            MediapipeGraphDefinition::status.handle(ovms::ValidationPassedEvent());
+        }
+
+        void mockMetadata() {
```
Comment (atobiszei) [inline src/test/get_mediapipe_graph_metadata_response_test.cpp:137]: to remove?

File: src/test/get_mediapipe_graph_metadata_response_test.cpp [hunk 2]
```diff
@@ -107,3 +122,207 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
...
+            return ovms::MediapipeGraphDefinition::status;
+        }
+    };
+
+    MockMediapipeGraphDefinitionGetInputsOutputsInfo graphDefinition;
+    KFSModelMetadataResponse response;
+    ConstructorEnabledModelManager manager;
+
+public:
+    void prepare() {
```
Comment (atobiszei) [inline src/test/get_mediapipe_graph_metadata_response_test.cpp:164]: To remove

File: src/test/get_mediapipe_graph_metadata_response_test.cpp [hunk 3]
```diff
@@ -107,3 +122,207 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
...
+        graphDefinition.mockMetadata();
+    }
+};
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, ModelVersionNotLoadedAnymoreButGraphNotReloadedYet) {
```
Comment (atobiszei) [inline src/test/get_mediapipe_graph_metadata_response_test.cpp:169]: MP doesn't use at all information about model used inside to produce it's metadata so we don't need that test.

File: src/test/get_mediapipe_graph_metadata_response_test.cpp [hunk 4]
```diff
@@ -107,3 +122,207 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
...
+    graphDefinition.mockStatus(ovms::StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE);
+    EXPECT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(graphDefinition, &response), ovms::StatusCode::OK);
+}
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, ModelVersionNotLoadedYet) {
```
Comment (atobiszei) [inline src/test/get_mediapipe_graph_metadata_response_test.cpp:174]: MP doesn't use at all information about model used inside to produce it's metadata so we don't need that test.

File: src/test/get_mediapipe_graph_metadata_response_test.cpp [hunk 5]
```diff
@@ -107,3 +122,179 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
...
+
+        cl_models_path = "/tmp/" + std::string(test_info->name());
+        cl_model_1_path = cl_models_path + "/model1/";
+        cl_model_2_path = cl_models_path + "/model2/";
+
+        const std::string FIRST_MODEL_NAME = "dummy";
+        const std::string SECOND_MODEL_NAME = "dummy_new";
+
+        std::filesystem::remove_all(cl_models_path);
+        std::filesystem::create_directories(cl_model_1_path);
```
Comment (atobiszei) [inline src/test/get_mediapipe_graph_metadata_response_test.cpp:207]: Either use TestWithTempDir or ensure cleanup of all tmp directories

File: src/test/get_mediapipe_graph_metadata_response_test.cpp [hunk 6]
```diff
@@ -107,3 +122,179 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
...
+    req.set_version("2");
+
+    std::unique_ptr<ServerShutdownGuard> serverGuard;
+    ovms::Server& server = ovms::Server::instance();
+    SetUpSingleModel(cl_models_path, "dummy");
+    auto& config = ovms::Config::instance();
+    auto retCode = server.startModules(config);
+    EXPECT_TRUE(retCode.ok()) << retCode.string();
+    if (!retCode.ok())
+        throw std::runtime_error("Failed to start modules");
```
Comment (atobiszei) [inline src/test/get_mediapipe_graph_metadata_response_test.cpp:280]: Why throw inside test?

File: ci/check_coverage.bat [hunk 7]
```diff
@@ -5,8 +5,8 @@ MIN_LINES_COV=76.8
 MIN_FUNCTION_COV=87.6
 
 #Rhel
-MIN_LINES_COV=75.9
-MIN_FUNCTION_COV=75.7
+MIN_LINES_COV=76.3
+MIN_FUNCTION_COV=75.3
```
Comment (atobiszei) [inline ci/check_coverage.bat:9]: Why lower function coverage?

Comment (rasapala) [reply ci/check_coverage.bat:9]: Yeah this is strange.

File: src/test/get_mediapipe_graph_metadata_response_test.cpp (hunk 6 above)
Comment (rasapala) [reply src/test/get_mediapipe_graph_metadata_response_test.cpp:280]: Removed.

File: src/test/get_mediapipe_graph_metadata_response_test.cpp (hunk 5 above)
Comment (rasapala) [reply src/test/get_mediapipe_graph_metadata_response_test.cpp:207]: Cleanup ensured with test TearDown cleanup

**Review by michalkulakowski — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]

**Pull request merged**